### DISCUSSION TITLE
Message receipt lookup

### DISCUSSION THREAD

**Discussion by jennijuju** [filecoin-project/lotus#9095] [Build and Sync]
With #9034, we are currently planning to have [splitstore](https://github.com/filecoin-project/lotus/blob/master/blockstore/splitstore/README.md) prune the datastore and discard all the states and message receipts after 4 finalities. Opening this discussion aims to gather feedback from tooling & application developers on whether this would impact their product if the node they are connecting enables split store.  i.e: do you need arbitrary receipt lookup using a lotus node? If so, what's the use case?

**Comment by jennijuju** [comment]
from @hannahhoward 

One thing that might be relevant is I don’t believe it’s possible to reconnect PieceCIDs to sector numbers without having messages, cause it’s the ProveCommit that links the deals to a sector number. This isn’t relevant in daily deal flow cause everything is stored already off chain in the PieceStore, but we have had issues where the PieceStore is deleted, and might need to reconstruct it. That’s a niche concern though.

**Reply by jennijuju** [reply]
from a market perspective ^

**Comment by Schwartz10** [comment]
Hey @jennijuju ! Summary - the Glif Explorer relies on being able to derive a Message Receipt from a signed message CID. We currently do this by `StateReplay`.

It seems like arbitrary receipt lookup would be helpful to us in convenience, but would it have any performance optimizations or additional info that we can't get from using StateReplay?

Also, what's the difference between arbitrary receipt lookup and https://github.com/filecoin-project/lotus/blob/master/api/v0api/full.go#L550 ?
